src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -354,20 +347,40 @@ export function DiscoverMainRoute({
   return (
     <DiscoverCustomizationProvider value={customizationService}>
       <DiscoverMainProvider value={stateContainer}>
-        <>
-          <DiscoverTopNavInline
-            stateContainer={stateContainer}
-            hideNavMenuItems={loading || showNoDataPage}
-          />
-          {mainContent}
-        </>
+        <DiscoverMainLoading
+          mainContent={mainContent}
+          showNoDataPage={showNoDataPage}
+          stateContainer={stateContainer}
+        />
       </DiscoverMainProvider>
     </DiscoverCustomizationProvider>
   );
 }
 // eslint-disable-next-line import/no-default-export
 export default DiscoverMainRoute;
 
+export function DiscoverMainLoading({
```

Comment:
since the loading state now is in the stateContainer we can need to have the state container in context so it's provided, therefore this change. This could be seen as first part of refactoring this file, which could use some love